src/Illuminate/Database/Eloquent/Relations/Relation.php

```diff
@@ -392,6 +396,24 @@ protected function whereInMethod(Model $model, $key)
                         : 'whereIn';
     }
 
+    /**
+     * Add a whereIn eager constraint for the given set of model keys to be loaded.
+     *
+     * @param  string  $whereIn
+     * @param  string  $key
+     * @param  array  $modelKeys
+     * @param  Builder  $query
+     * @return void
+     */
+    protected function whereInEager(string $whereIn, $key, array $modelKeys, $query = null)
+    {
+        ($query ?? $this->query)->{$whereIn}($key, $modelKeys);
+
+        if ($modelKeys === []) {
+            $this->eagerImplicitlyEmpty = true;
+        }
```

Comment:
Well if the eager is loaded again it'll make calls to the relation methods `addEagerConstraints` which subsequently call `whereInEager` which could change the `eagerImplicitlyEmpty` property because it will know if there are any loadable keys. So once the eager constraints are added, it will set that property.